File: vol2/ch22/LeetCode1196.py
class Solution:
    def maxNumberOfApples(self, weight: List[int]) -> int:
        n=len(weight)
        weight.sort()   #递增排序
        ans,r=0,5000
        for i in range(0,n):
            if weight[i]<=r:
                ans+=1
                r-=weight[i]
            else: break
        return ans

File: vol2/ch22/LeetCode1196.cpp
class Solution {
public:
    int maxNumberOfApples(vector<int>& weight) {
        int n=weight.size();
        sort(weight.begin(),weight.end());  //µ›‘ˆ≈≈–Ú
        int ans=0;
        int r=5000;
        for(int i=0;i<n;i++) {
            if(weight[i]<=r) {
                ans++;
                r-=weight[i];
            }
            else break;
        }
        return ans;
    }
};

